charms/slurmd/src/utils/nhc.py

```
@@ -0,0 +1,123 @@
+
+    Raises:
+        subprocess.CalledProcessError: Raised if error is encountered during nhc install.
+    """
+    _logger.info("installing node health check (nhc)")
+    with tempfile.TemporaryDirectory() as tmpdir:
+        try:
+            env = {"LC_ALL": "C", "LANG": "C.UTF-8"}
+
+            _logger.info("extracting nhc tarball")
+            r = subprocess.check_output(
+                [
+                    "tar",
+                    "--extract",
+                    "--directory",
+                    tmpdir,
+                    "--file",
+                    "lbnl-nhc-1.4.3.tar.gz",
+                    "--strip",
+                    "1",
+                ],
+                stderr=subprocess.STDOUT,
+                text=True,
+            )
+            _logger.debug(r)
+
+            _logger.info("building nhc with autotools")
+            r = subprocess.check_output(
+                ["./autogen.sh", "--prefix=/usr", "--sysconfdir=/etc", "--libexecdir=/usr/lib"],
+                cwd=tmpdir,
+                env=env,
+                stderr=subprocess.STDOUT,
+                text=True,
+            )
+            _logger.debug(r)
+
+            _logger.info("testing nhc build")
+            r = subprocess.check_output(
+                ["make", "test"], cwd=tmpdir, env=env, stderr=subprocess.STDOUT, text=True
+            )
+            _logger.debug(r)
+
+            _logger.info("installing nhc")
+            r = subprocess.check_output(
+                ["make", "install"], cwd=tmpdir, env=env, stderr=subprocess.STDOUT, text=True
+            )
+            _logger.debug(r)
+        except subprocess.CalledProcessError as e:
+            _logger.error("failed to install nhc. reason: %s", e)
+            raise
+
+
+def get_config() -> str:
+    """Get the current NHC configuration."""
+    target = Path("/etc/nhc/nhc.conf")
+    if target.exists():
+        return target.read_text()
+
+    return f"{target} not found."
```

Raise an error instead of returning a plain string?

---

No error is raised because `nhc.get_config()` only has one usage which is in the `show_nhc_config` action handler:

https://github.com/charmed-hpc/hpc-libs/blob/eaa1978d2c51f22e6bffd6a1cea56e54b67534d4/lib/charms/hpc_libs/v0/slurm_ops.py#L911-L921

The method will succeed either way, but will let whoever runs the action know that _nhc.conf_ does not exist. I can change the function to instead bubble up a `FileNotFoundError`, but we should further discuss if we want to change the behavior of the `show_nhc_config` action itself to raise a non-zero exit code if _nhc.conf_ is not found.